### ISSUE TITLE
Reading 10000 rows of data takes more than 3 minutes

### ISSUE THREAD

**Issue by msarkrish** [simolus3/moor#2435]
I'm trying to read 10,000 data from the table. Drift taking almost more than 3 minutes. 

    drift: ^2.7.0

I'm using flutter 3.10.1 latest version

    Future<List<Asset>> getAssetsList({required String siteID}) async {
    final selectableQueryRow = customSelect(
        'select *, case when asset.location is not'
        ' null then (select description from locations where '
        'location = asset.location) else null end as locationdesc from '
            "asset where siteid='$siteID'");
    final queryRowList = await selectableQueryRow.get();
    final assetList = <Asset>[];
    for (final queryRow in queryRowList) {
        final assetData = queryRow.data;
        final asset = Asset(
            assetID: assetData['assetnum'] as String,
            assetDescription: assetData['description'] as String?,
            assetLocationDescription: assetData['locationdesc'] as String?,
            assetLocationID: assetData['location'] as String,
         );
         assetList.add(asset);
     }
    return assetList;
    }

**Comment by simolus3** [comment]
Did you resolve this issue? Out of interest, what was the problem?

**Comment by msarkrish** [comment]
Sorry @simolus3 for not updating the reason while closing the issue. Actually the problem is with query. I have used case statement for selecting value from second table(location) that takes more time. 

After I have modified the query with join, It is fetching data's immediately from table. 

This is my updated query. 
    
    select *,locations.description  from asset inner join locations on asset.location=locations.location where asset.siteid=''

Thank you for Asking @simolus3
